### PR TITLE
Refactor context checks (follow up from #245)

### PR DESCRIPTION
**Pull request by roncohen**

### CODE REVIEW

File: packages/node-sdk/src/client.ts [hunk 1]
```diff
@@ -395,7 +396,7 @@ export class BucketClient {
    *
    * @param context - The context to update.
    */
-  private async syncContext(options: ContextWithTracking) {
+  private async syncContext(options: { enableTracking: boolean } & Context) {
```
[the remark this answers is not in the document]
Comment (pavkam) [reply packages/node-sdk/src/client.ts:399]: Works as well

File: packages/node-sdk/src/client.ts [hunk 2]
```diff
@@ -1090,28 +1077,26 @@ export class BoundBucketClient {
   }
 }
 
-function checkContextWithTracking(context: ContextWithTracking) {
+function checkContextWithTracking(
+  context: ContextWithTracking,
+): asserts context is ContextWithTracking & { enableTracking: boolean } {
```
Comment (pavkam) [inline packages/node-sdk/src/client.ts:1082]: Nice!

File: packages/node-sdk/src/utils.ts [hunk 3]
```diff
@@ -14,6 +14,13 @@ export function ok(condition: boolean, message: string): asserts condition {
   }
 }
 
+export function assertId(id: string | number, entity: string) {
```
Comment (pavkam) [inline packages/node-sdk/src/utils.ts:17]: Maybe add an optional param `allowUndefined`. Would remove the need for more ifs in code above

Comment (pavkam) [inline packages/node-sdk/src/utils.ts:17]: Nit: To keep consistent maybe name it idOk

File: packages/node-sdk/src/client.ts [hunk 4]
```diff
@@ -539,13 +536,9 @@ export class BucketClient {
    **/
   public async updateCompany(
     companyId: string | number,
-    opts?: TrackOptions & { userId?: string },
+    opts?: TrackOptions & { userId?: string | number },
```
Comment (pavkam) [inline packages/node-sdk/src/client.ts:539]: Still think we can make a type for these instead of repeating it.

**Review by pavkam — APPROVED** [review]

File: packages/node-sdk/src/client.ts (hunk 4 above)
Comment (roncohen) [reply packages/node-sdk/src/client.ts]: done 👍

File: packages/node-sdk/src/utils.ts (hunk 3 above)
Comment (roncohen) [reply packages/node-sdk/src/utils.ts]: renamed!

Comment (roncohen) [reply packages/node-sdk/src/utils.ts]: I will leave as it. Hate boolean args ;)

**Pull request merged**